modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbStorage.java

```diff
@@ -69,56 +96,131 @@ public class TxStateRocksDbStorage implements TxStateStorage, AutoCloseable {
     /** Database options. */
     private volatile TransactionDBOptions txDbOptions;
 
+    /** Write options. */
+    private final WriteOptions writeOptions = new WriteOptions().setDisableWAL(true);
+
+    /** Read options for regular reads. */
+    private final ReadOptions readOptions = new ReadOptions();
+
+    /** Read options for reading persisted data. */
+    private final ReadOptions persistedTierReadOptions = new ReadOptions().setReadTier(PERSISTED_TIER);
+
     /** Thread-pool for snapshot operations execution. */
-    private final ExecutorService snapshotExecutor;
+    private final ExecutorService threadPool;
 
-    /** Snapshot manager. */
-    private volatile RocksSnapshotManager snapshotManager;
+    /** On-heap-cached last applied index value. */
+    private volatile long lastAppliedIndex;
 
-    /** Snapshot restore lock. */
-    private final Object snapshotRestoreLock = new Object();
+    /** The value of {@link #lastAppliedIndex} persisted to the device at this moment. */
+    private volatile long persistedIndex;
 
     /** Whether is started. */
     private boolean isStarted;
 
+    /** Collection of opened RocksDB iterators. */
+    private final Set<RocksIterator> iterators = ConcurrentHashMap.newKeySet();
+
+    /** Busy lock to stop synchronously. */
+    final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    /** Prevents double stopping the component. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
     /**
      * The constructor.
      *
      * @param dbPath Database path.
-     * @param snapshotExecutor Snapshot thread pool.
+     * @param threadPool Thread pool.
      */
-    public TxStateRocksDbStorage(Path dbPath, ExecutorService snapshotExecutor) {
+    public TxStateRocksDbStorage(Path dbPath, ExecutorService threadPool) {
         this.dbPath = dbPath;
-        this.snapshotExecutor = snapshotExecutor;
+        this.threadPool = threadPool;
     }
 
     /** {@inheritDoc} */
     @Override public void start() {
         try {
-            this.options = new Options().setCreateIfMissing(true);
+            flusher = new RocksDbFlusher(
+                    busyLock,
+                    //TODO Use shared pool.
+                    newSingleThreadScheduledExecutor(new NamedThreadFactory("tx-state-store-scheduled-pool", LOG)),
```

Comment:
Should we just add another param in addition to threadPool one?

Reply:
Yes, I totally forgot about it